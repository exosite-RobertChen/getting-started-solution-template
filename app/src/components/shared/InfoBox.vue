<template>
  <v-row
    :class="{ 'subtitle-1': $vuetify.breakpoint.xs }"
    class="headline fill-height"
    justify="center"
    align="center"
  >
    <v-card
      :class="{
        'px-6': $vuetify.breakpoint.xs,
        'px-10': !$vuetify.breakpoint.xs
      }"
      class="center py-10 text-center elevation-5 border-radius-1"
    >
      <div
        v-for="text in texts"
        :key="text"
        class="mb-2"
        :class="{ 'accent--text': error }"
      >
        {{ text }}
      </div>
      <div v-if="btnText">
        <span>Go ahead and</span>
        <v-btn
          class="ml-3"
          color="accent"
          :small="$vuetify.breakpoint.xs"
          data-qa="InfoBoxButton"
          @click="$emit('action')"
          >{{ btnText }}</v-btn
        >
      </div>
      <div class="mt-6">
        <a
          :href="links.help"
          class="subtitle-1 info-box__link"
          target="_blank"
          primary--text
          >Need help?</a
        >
      </div>
    </v-card>
  </v-row>
</template>

<script>
export default {
  name: 'InfoBox',
  props: {
    texts: {
      type: Array,
      required: true
    },
    btnText: {
      type: String,
      default: ''
    },
    links: {
      type: Object,
      required: true
    },
    error: {
      type: Boolean
    }
  }
}
</script>

<style lang="scss">
.info-box {
  &__link {
    text-decoration: underline;
  }
}
</style>
