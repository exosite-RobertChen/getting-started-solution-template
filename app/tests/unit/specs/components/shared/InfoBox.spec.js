import merge from 'lodash.merge'
import { shallowComponent } from '@/../tests/unit/common'
import InfoBox from '@/components/shared/InfoBox.vue'
import branding from '@/../vendor/branding'

const defaultProps = {
  propsData: {
    texts: ['This is a sample text'],
    btnText: 'Claim Device',
    links: branding.links,
    error: false
  }
}

const factory = overrides =>
  shallowComponent(InfoBox, merge(defaultProps, overrides))

describe('InfoBox.vue', () => {
  test('can be mounted', () => {
    const wrapper = factory()
    expect(wrapper.contains(InfoBox)).toBe(true)
  })

  test('matches snapshot', () => {
    const wrapper = factory()

    expect(wrapper.element).toMatchSnapshot()
  })

  test('renders a button when button text supplied', () => {
    const wrapper = factory()
    expect(wrapper.contains('[data-qa=InfoBoxButton]')).toBe(true)
  })

  test(`doesn't render a button when button text not supplied`, () => {
    const wrapper = factory({
      propsData: {
        btnText: ''
      }
    })
    expect(wrapper.contains('[data-qa=InfoBoxButton]')).toBe(false)
  })
})
