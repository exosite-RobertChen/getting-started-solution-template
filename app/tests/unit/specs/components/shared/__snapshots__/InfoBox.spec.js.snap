// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBox.vue matches snapshot 1`] = `
<v-row-stub
  align="center"
  class="headline fill-height [object Object]"
  justify="center"
  tag="div"
>
  <v-card-stub
    class="center py-10 text-center elevation-5 border-radius-1 px-10"
    loaderheight="4"
    tag="div"
  >
    <div
      class="mb-2"
    >
      
      This is a sample text
    
    </div>
     
    <div>
      <span>
        Go ahead and
      </span>
       
      <v-btn-stub
        activeclass=""
        class="ml-3"
        color="accent"
        data-qa="InfoBoxButton"
        tag="button"
        type="button"
      >
        Claim Device
      </v-btn-stub>
    </div>
     
    <div
      class="mt-6"
    >
      <a
        class="subtitle-1 info-box__link"
        href="https://docs.exosite.com/connectivity/pdaas/device-owner/"
        primary--text=""
        target="_blank"
      >
        Need help?
      </a>
    </div>
  </v-card-stub>
</v-row-stub>
`;
